python_modules/dagster-graphql/dagster_graphql/implementation/utils.py

```diff
@@ -21,6 +18,15 @@ def _fn(self, graphene_info, *args, **kwargs):  # pylint: disable=unused-argumen
     return decorator
 
 
+def assert_permission(graphene_info, permission):
+    from dagster_graphql.schema.errors import GrapheneUnauthorizedError
+
+    if not graphene_info.context.has_permission(permission):
+        raise UserFacingGraphQLError(GrapheneUnauthorizedError())
+
+    return True
```

Comment:
I see what you're going for with the

'if condition and assert_permission(...)' pattern, my instinct is that returning nothing here is clearer though (so that people don't incorrectly think they need to handle the return False case

Reply:
That's a reasonable point. I ended up structuring it this way based on @alangenfeld's comment on the other PR, 
> ... Might be clearer to have the 2 perms checks in an if / else instead of this assumption/precedence set up...

My thought was that having an imperative statement:
```python
if need_for_perm:
  assert_permission(graphene_info, "permission_for_this_condition")
  perform_action()
```
is less visually obvious that a permissions check is happening, whereas a guard structure makes the check more clear:
```python
if need_for_perm and assert_permission(graphene_info, "permission_for_this_condition"):
  perform_action()
```
Here, it's more clear that the indented code block can only run if the user has satisfied the given conditions. In the first case, the block is entered even if the permission doesn't hold (until the imperative assertion is hit).

One alternative could be just to have a `has_permission` utility function, but then every use would need to throw the perms error itself instead of having the util method do it for them.

Reply:
> is less visually obvious that a permissions check is happening

To me the `assert_` / `verify_` / `check_` prefix means "this function is doing work to make assertions and will throw if something is not right" so the `-> None` variant reads fine to me. We use this pattern around the codebases, with `check_` i think being the predominant prefix.